dbcon/joblist/tuple-bps.cpp

```diff
@@ -2672,15 +2682,15 @@ void TupleBPS::receiveMultiPrimitiveMessages(uint32_t threadID)
 
         if (ffirstStepType == SCAN && bop == BOP_AND && !cancelled())
         {
-            cpMutex.lock();
+            //cpMutex.lock();
             lbidList->UpdateAllPartitionInfo(fColType);
-            cpMutex.unlock();
+            // cpMutex.unlock();
         }
     }
 
     // Bug 3136, let mini stats to be formatted if traceOn.
-    if (lastThread && !didEOF)
-        dlp->endOfInput();
+    // How the comment above is related to the next line of code?
+    dlp->endOfInput();
```

Comment:
We should move this line up the code into 2621 so we notify the higher SQL operators earlier. After 2621 there is only stats related ops.

Reply:
Or even to 2579.